docs/style.md

```diff
@@ -0,0 +1,223 @@
+              if (NULL == (y = malloc(10))
+              {
+              }
+            }
+
+    - Don't:
+            if ((7 == a) && (NULL == (y = malloc(10))))
+            {
+            }
+
+
+- Avoid ternary operators with complex statements
+- Avoid deeply nested if statements: use if-else chains, helper functions, early returns or goto to cleanup code:
+    - Do:
+
+            void someFunction(int x)
+            {
+              int result = 0;
+              char* foo = NULL;
+              char* bar = NULL;
+              char* baz = NULL;
+
+              if (7 == x)
+              {
+                return EINVAL;
+              }
+
+              foo = func();
+              if (NULL == foo)
+              {
+                result = ENOMEM;
+                goto cleanup;
+              }
+
+              bar = func();
+              if (NULL == bar)
+              {
+                result = ENOMEM;
+                goto cleanup;
+              }
+
+              baz = func();
+              if (NULL == baz)
+              {
+                result = ENOMEM;
+                goto cleanup;
+              }
+
+              result = doSomething(foo, bar, baz);
+
+            cleanup:
+              if (NULL != foo)
+              {
+                free(foo);
+              }
+              if (NULL != bar)
+              {
+                free(bar);
+              }
+              if (NULL != baz)
+              {
+                free(baz);
+              }
+              return result;
+            }
+
+    - Don't:
+
+            void someFunction(int x)
+            {
+              int result = 0;
+              char* foo = NULL;
+              char* bar = NULL;
+              char* baz = NULL;
+              foo = func();
+              if (7 == x) {
+                if (NULL != foo)
+                {
+                  bar = func();
+                  if (NULL != bar)
+                  {
+                    baz = func();
+                    if (NULL != bar)
+                    {
+                      result = doSomething(foo, bar, baz);
+                    }
+                    else
+                    {
+                      result = ENOMEM;
+                    }
+                  }
+                  else
+                  {
+                    result = ENOMEM;
+                  }
+                }
+                else
+                {
+                  result = ENOMEM;
+                }
+              }
+              else
+              {
+                result = EINVAL;
+              }
+
+              if (NULL != foo)
+              {
+                free(foo);
+              }
+              if (NULL != bar)
+              {
+                free(bar);
+              }
+              if (NULL != baz)
+              {
+                free(baz);
+              }
+              return result;
+            }
+
+
+# Naming Conventions
+- PascalCase for files, functions, types, and structs
```

Comment:
>files, functions, types, and structs [](http://example.com/codeflow?start=17&length=36)

Consider changing this line as following, to complete names and list:

```
- PascalCase for file names, function names, types (including class and struct), and namespaces.
```

We need the namespaces added here, also struct is a type and we also need there at least class added.